internal/terraform/json_test.go

```diff
@@ -135,27 +135,29 @@ func TestResourceProviderConfig(t *testing.T) {
 	p := unmarshalTestPlan(t)
 
 	tests := []struct {
-		kind    string
-		name    string
-		wantErr bool
+		kind      string
+		name      string
+		wantEmpty bool
 	}{
-		// Empty address - should return err.
+		// Empty address - should return empty.
 		{"", "", true},
 
-		// No provider for this resource - should return err.
+		// No provider for this resource - should return empty.
 		{"google_storage_bucket", "some_other_bucket", true},
 
-		// Provider config exists - should not return err.
+		// Provider config exists - should not return empty.
 		{"google_storage_bucket", "gcs_tf_bucket", false},
 	}
 	for _, testcase := range tests {
-		config, err := resourceProviderConfig(testcase.kind, testcase.name, p)
-		if testcase.wantErr {
-			if err == nil {
-				t.Errorf("resourceProviderConfig(%q, %q, %v) returned nil err; wanted err != nil", testcase.kind, testcase.name, testPlanPath)
+		config, ok := resourceProviderConfig(testcase.kind, testcase.name, p)
+		isEmpty := cmp.Equal(config, ProviderConfig{}, cmpopts.EquateEmpty())
```

Comment:
You shouldn't need to check empty anymore. The ok bool should be the only thing you need to check and if ok then do cmp.Diff

Reply:
I thought about this, I think of it as double-checking - if it returns !ok, it should actually also be returning an empty config, so on the chance that someone ignores the ok and just uses the returned config, it's still not going to be partially populated. WDYT?

Reply:
Most go code does not support using the return value if there is a corresponding ok/err field. Given this is a private func it's even less of a concern. I wouldn't bother and simplify code/tests if possible.

Reply:
I think I see what you mean about using ok and diff. I've simplified the test now.